Source/buildbindingccpp.go

```diff
@@ -1389,6 +1398,41 @@ func buildCppHeader(component ComponentDefinition, w LanguageWriter, NameSpace s
 		w.Writeln("};")
 	}
 
+	w.Writeln("")
+	w.Writeln("/*************************************************************************************************************************")
+	w.Writeln(" RTTI static getClassName implementations")
+	w.Writeln("**************************************************************************************************************************/")
+	w.Writeln("")
+
+	for i := 0; i < len(component.Classes); i++ {
+		class := component.Classes[i]
+		w.Writeln("  const std::string &C%s::getClassName()", class.ClassName)
+		w.Writeln("  {")
+		w.Writeln("  static const std::string s_sClassName = \"%s\";", class.ClassName)
+		w.Writeln("  return s_sClassName;")
+		w.Writeln("  }")
+		w.Writeln("")
+	}
+
+	w.Writeln("")
+	w.Writeln("/*************************************************************************************************************************")
+	w.Writeln(" RTTI cast implementation")
+	w.Writeln("**************************************************************************************************************************/")
+	w.Writeln("")
+
+	w.Writeln("  template <class T>")
+	w.Writeln("  std::shared_ptr<T> %s_cast(PBase pObj)", strings.ToLower(NameSpace))
+	w.Writeln("  {")
+	w.Writeln("    static_assert(std::is_convertible<T, CBase>::value, \"T must be convertible to %s::CBase\");", NameSpace)
+	w.Writeln("")
+	w.Writeln("    if (pObj && pObj->m_pWrapper->ImplementsInterface(pObj.get(), T::getClassName())){")
+	w.Writeln("      return std::make_shared<T>(pObj->m_pWrapper, pObj->m_pHandle);")
```

Comment:
Need to call `pObj->wrapper()->AcquireInstance(pObj);` to make sure reference count is correct.